src/main/app/controller/UploadController.ts

```diff
@@ -0,0 +1,448 @@
+          res.redirect(this.getNextPageRedirectUrl());
+        } catch (error) {
+          this.createUploadedFileError(req, res, chooseFileLink, 'UPLOAD_DELETE_FAIL_ERROR');
+        }
+      }
+    }
+  }
+
+  public UploadDocumentInstance(BASEURL: string, header: RawAxiosRequestHeaders): AxiosInstance {
+    return axios.create({
+      baseURL: BASEURL,
+      headers: header,
+      maxContentLength: Infinity,
+      maxBodyLength: Infinity,
+    });
+  }
+
+  private getOtherInfoDocuments(req: AppRequest<AnyObject>): Document[] {
+    return req.session['otherCaseInformation'].map(document => {
+      // eslint-disable-next-line @typescript-eslint/no-shadow
+      const { url, fileName, documentId, binaryUrl } = document;
+      return {
+        id: documentId,
+        value: {
+          documentLink: {
+            document_url: url,
+            document_filename: fileName,
+            document_binary_url: binaryUrl,
+          },
+        },
+      };
+    });
+  }
+
+  private getSupportingDocuments(req: AppRequest<AnyObject>): Document[] {
+    return req.session['supportingCaseDocuments'].map(document => {
+      // eslint-disable-next-line @typescript-eslint/no-shadow
+      const { url, fileName, documentId, binaryUrl } = document;
+      return {
+        id: documentId,
+        value: {
+          documentLink: {
+            document_url: url,
+            document_filename: fileName,
+            document_binary_url: binaryUrl,
+          },
+        },
+      };
+    });
+  }
+
+  private getTribunalFormDocuments(req: AppRequest<AnyObject>): Document[] {
+    return req.session['caseDocuments'].map(document => {
+      const { url, fileName, documentId, binaryUrl } = document;
+      return {
+        id: documentId,
+        value: {
+          documentLink: {
+            document_url: url,
+            document_filename: fileName,
+            document_binary_url: binaryUrl,
+          },
+        },
+      };
+    });
+  }
+
+  private uploadFileError(
+    req: AppRequest<AnyObject>,
+    res: Response<any, Record<string, any>>,
+    redirectUrlIfError: string,
+    errorMessage?: string,
+    link?: string
+  ): void {
+    req.session.fileErrors.push({
+      text: errorMessage,
+      href: link,
+    });
+
+    this.redirect(req, res, redirectUrlIfError);
+  }
+
+  /**
+   *
+   * @param req
+   * @param res
+   */
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    await this.submit(req, res);
+  }
+
+  /**
+   *
+   * @param req
+   * @param res
+   */
+  public async submit(req: AppRequest<AnyObject>, res: Response<any, Record<string, any>>): Promise<void> {
+    const chooseFileLink = '#file-upload-1';
+    const filesUploadedLink = '#filesUploaded';
+    const { documentUploadProceed } = req.body;
+
+    if (this.shouldSetUpFormData()) {
+      this.setUpForm(req);
+    }
+
+    let TotalUploadDocuments = 0;
```

Comment:
can we follow best practice with variable names - camelCase - please.

Reply:
@tomxelliott this has been updated, please could you review again?